backend/src/main/java/com/myshop/products/Product.java

```diff
@@ -1,126 +1,106 @@
 package com.myshop.products;
 
-import com.myshop.utils.SLUGIFY;
-
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.myshop.categories.Category;
+import lombok.Getter;
+import lombok.Setter;
 import javax.persistence.*;
-import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
+import java.sql.Date;
 import java.util.List;
 
 @Entity
-@Table( name = "products",
-        indexes =@Index(columnList = "name_id", unique = true))
-
+@Setter
+@Getter
+@Table(name = "products",
+        indexes = {
+                @Index(columnList = "id", unique = true),
+                @Index(columnList = "name")
+        })
 public class Product {
     @Id
     @GeneratedValue(strategy = GenerationType.SEQUENCE)
     @Column(name = "id", nullable = false)
     private Long id;
-    @Column(name = "name_id", nullable = false)
-    @NotBlank(message = "Name is required")
-    private String nameId;
-    @NotNull
+    @NotNull(message = "Name is required")
     private String name;
-    @NotNull
+    @NotNull(message = "Price is required")
     private float price;
-    @NotNull
+    @NotNull(message = "ImageURL is required")
     private String imageUrl;
     @ElementCollection
     private List<String> color;
     @ElementCollection
     private List<String> size;
     private String description;
+    @ManyToOne(fetch = FetchType.EAGER)
+    private Category category;
```

Comment:
the created_at, updated_at and deleted_at seem like will be reused for other entities. can we do something to avoid repeating this step.

Reply:
- I had found the solution that using @MappedSuperclass  annotation for superclass which includes repeated attributes. And then use inheritance for class that contain those attributes. 
- Reference: **https://rieckpil.de/howto-avoid-repeating-attributes-in-jpa-entities/**